### DISCUSSION TITLE
Does BBT generate zotero:// URIs?

### DISCUSSION THREAD

**Discussion by glocalglocal** [retorquere/zotero-better-bibtex#2382] [Q&A]
URIs in the form `zotero://select/items/@my_citationkey` work well in getting Zotero to locate and focus on an item. I can generate those manually and Citations for Obsidian makes them available as `ZoteroSelecttURI` but is there a way to get Zotero or BBT to generate such a link?

Zutilo generates URIs but the format is different and it points to the non-BBT citation keys -- eg `zotero://select/library/items/TXLG58K4`. For that reason, I am guessing only BBT would be able to handle the citation keys it generates.

**Comment by retorquere** [comment]
> is there a way to get Zotero or BBT to generate such a link?

I'm not aware of a way to do this.

> non-BBT citation keys -- eg zotero://select/library/items/TXLG58K4

Those aren't citation keys, they're the internal keys Zotero uses.

**Reply by glocalglocal** [reply]
> > is there a way to get Zotero or BBT to generate such a link?
> 
> I'm not aware of a way to do this.

I take it this means BBT doesn't do it at the moment. As you are the maintainer, would you perhaps consider adding this as a menu option? Generating the URI manually is prone to user errors.

**Reply by retorquere** [reply]
Where in the Zotero UI does Zutilo allow the user to generate these links?

**Reply by glocalglocal** [reply]
> Where in the Zotero UI does Zutilo allow the user to generate these links?

In the Zutilo context menu: `Copy select item links` puts the Zotero key URI into the clipboard in this form: `zotero://select/library/items/TXLG58K4`. I was suggesting that BBT could do something similar with `zotero://select/items/@my_citationkey`. Otherwise, we have to construct the URI manually, which is prone to errors.

**Comment by retorquere** [comment]
> URIs in the form `zotero://select/items/@my_citationkey` work well in getting Zotero to locate and focus on an item. I can generate those manually and Citations for Obsidian makes them available as `ZoteroSelecttURI` but is there a way to get Zotero or BBT to generate such a link?
> 
> Zutilo generates URIs but the format is different and it points to the non-BBT citation keys -- eg `zotero://select/library/items/TXLG58K4`. For that reason, I am guessing only BBT would be able to handle the citation keys it generates.

That depends on what you mean "handle". As long as BBT is installed, `zotero://...@itemkey` will work wherever `zotero://...<itemkey>` works, even for non-BBT functionality. So non-BBT functionality can "handle" `zotero://...@itemkey` links.

**Reply by glocalglocal** [reply]
By 'handle' I meant construct `zotero://...@itemkey` for an item and place it into the clipboard -- ie do what Zutilo does above.
